common/crypto/dkg/src/dealing.rs

```diff
@@ -0,0 +1,485 @@
+// Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::bte::proof_chunking::ProofOfChunking;
+use crate::bte::proof_sharing::ProofOfSecretSharing;
+use crate::bte::{
+    encrypt_shares, proof_chunking, proof_sharing, Ciphertexts, Epoch, Params, PublicKey,
+};
+use crate::error::DkgError;
+use crate::interpolation::polynomial::{Polynomial, PublicCoefficients};
+use crate::interpolation::{
+    perform_lagrangian_interpolation_at_origin, perform_lagrangian_interpolation_at_x,
+};
+use crate::{NodeIndex, Share, Threshold};
+use bls12_381::{G2Projective, Scalar};
+use rand_core::RngCore;
+use std::collections::BTreeMap;
+use zeroize::Zeroize;
+
+#[derive(Debug)]
+#[cfg_attr(test, derive(PartialEq))]
+pub struct Dealing {
+    pub public_coefficients: PublicCoefficients,
+    pub ciphertexts: Ciphertexts,
+    pub proof_of_chunking: ProofOfChunking,
+    pub proof_of_sharing: ProofOfSecretSharing,
+}
+
+impl Dealing {
+    // I'm not a big fan of this function signature, but I'm not clear on how to improve it while
+    // allowing the dealer to skip decryption of its own share if it was also one of the receivers
+    pub fn create(
```

Comment:
In the `create` function we should also add as an optional input argument the secret value (i.e., the constant value of the polynomial) known by the validator who is calling this function. This secret value will be useful if we want to do a re-sharing of the same secret and verification key.